functions/snippets/src/main/java/FileSystem.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START functions_concepts_filesystem]
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.nio.file.Files;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class FileSystem {
+
+  // Lists the files in the current directory.
+  public void listFiles(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+
+    String currentDirectory = System.getProperty("user.dir");
+    Path path = Paths.get(currentDirectory, "snippets/src/main/java/");
```

Comment:
Taking a second look at this, I think this will actually fail if the user tried to deploy it since the source won't be available. The Go sample uses the current working directory, so can you try this instead?

```java
  // Lists the files in the current directory.
  public void listFiles(HttpServletRequest request, HttpServletResponse response) {
     File cur = new File(".");
     File[] files = cur.listFiles();

     PrintWriter writer = response.getWriter();
     writer.write("Files: \n");
     for (File f : files) {
          writer.write("\t" + f + "\n");
     } 
  }
```
The test can probably just get away with checking for `"Files: "`(since cwd will change depending on environment)

Reply:
Good point. The previous implementation required that file structure.

I've changed the implementation to adhere to your suggestion.